src/tasks/libraries-db.js

```diff
@@ -237,6 +246,28 @@ function syncLibraryVersions(target, record, stateMap) {
         });
 }
 
+function compareVersions(a, b) {
+    var BRANCHES = ['master', 'dev'],
+        VERSION_REGEXP = /^\d+\.\d+\.\d+$/;
+
+    if (BRANCHES.indexOf(a) !== -1) { return 1; }
+    if (BRANCHES.indexOf(b) !== -1) { return -1; }
+
+    a = semver.clean(a);
+    b = semver.clean(b);
+
+    if (VERSION_REGEXP.test(a) && VERSION_REGEXP.test(b)) { return semver.rcompare(a, b); }
+
+    if (VERSION_REGEXP.test(a)) { return -1; }
+    if (VERSION_REGEXP.test(b)) { return 1; }
+
+    if (semver.valid(a) && semver.valid(b)) { return semver.rcompare(a, b); }
```

Comment:
I don't remember all details, but I can garantee that this code gives a valid result.
I was tested by time